drivers/flash/spi_nor.c

```diff
@@ -110,6 +135,87 @@ static int spi_nor_access(const struct device *const dev,
 #define spi_nor_cmd_addr_write(dev, opcode, addr, src, length) \
 	spi_nor_access(dev, opcode, true, addr, (void *)src, length, true)
 
+static int enter_dpd(const struct device *const dev)
+{
+	int ret = 0;
+
+	if (IS_ENABLED(DT_INST_0_JEDEC_SPI_NOR_HAS_DPD)) {
+		ret = spi_nor_cmd_write(dev, SPI_NOR_CMD_DPD);
+		if (ret == 0) {
+#ifdef DT_INST_0_JEDEC_SPI_NOR_T_ENTER_DPD
+			k_sleep(K_MSEC(T_DP_MS));
+#endif /* T_ENTER_DPD */
+		}
+	}
+	return ret;
+}
+
+static int exit_dpd(const struct device *const dev)
+{
+	int ret = 0;
+
+	if (IS_ENABLED(DT_INST_0_JEDEC_SPI_NOR_HAS_DPD)) {
+#ifdef DT_INST_0_JEDEC_SPI_NOR_DPD_WAKEUP_SEQUENCE
+		/* Delay for tDPDD */
+		k_sleep(K_MSEC(T_DPDD_MS));
+		/* Assert CSn and wait for tCRDP.
+		 *
+		 * Unfortunately the SPI API doesn't allow us to
```

Comment:
Yes.  This is certainly shaky, since depending on the delay and the bus frequency the timing requirement might not be met, but without control of CSn directly I don't have an alternative.  (We can't just grab the GPIO and muck with it because some drivers don't provide CSn GPIO information.)  It's also possible but unlikely that sending the command would cause the chip to misbehave.  I could see substituting `RDID` instead.

Experimentally this works on at least two devices that have this (efr32mg_sltb002a's MX25R80 support isn't enabled in tree yet, but I've tested it).

Reply:
Switched to using `RDID`, just in case.